### PR TITLE
Backporting changes from build.py to package.sh

### PR DESCRIPTION
**Pull request by rossmcdonald**
A few changes to packaging/build scripts:
* Added RC and iteration support for `build.py`
* Re-added Linux ARM as a build target to `build.py`
* Removed `post-install` function from `package.sh` and moved to it's own script in `scripts/`
* Added pre/post install/uninstall scripts to `package.sh`

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -88,9 +88,9 @@
     'darwin': [ "amd64", "386" ],
     # Windows is not currently supported in InfluxDB 0.9.5 due to use of mmap
     # 'windows': [ "amd64", "386", "arm" ],
-    'linux': [ "amd64", "386" ]
+    'linux': [ "amd64", "386", "arm" ]
 }
-supported_go = [ '1.5.1' ]
+supported_go = [ '1.5.1', '1.4.2' ]
```
Comment (otoolep) [inline build.py]: What is the idea behind this list? A check?

**Comment by otoolep** [comment]
+1, and one question.

File: build.py (hunk 1 above)
Comment (rossmcdonald) [inline build.py]: Yes. This isn't used yet, but I was going to compare version numbers to either prevent builds on unsupported Go versions, or allow for multi-Go-version builds at some point in the future.

**Comment by otoolep** [comment]
This will need to be cherry-picked to the 0.9.5 branch.

File: scripts/post-install.sh [hunk 2]
```diff
@@ -8,7 +8,6 @@ LOGROTATE_DIR=/etc/logrotate.d
 if ! id influxdb >/dev/null 2>&1; then
         useradd --system -U -M influxdb -s /bin/false -d $DATA_DIR
 fi
-chown influxdb:influxdb $BIN_DIR/influx*
```
Comment (rossmcdonald) [inline scripts/post-install.sh]: @nathanielc Per our conversation about binary permissions.

**Comment by rossmcdonald** [comment]
@otoolep Not a problem. I also made a change to the binary owner/group.

File: scripts/post-install.sh (hunk 2 above)
Comment (nathanielc) [inline scripts/post-install.sh]: thanks LGTM

Comment (otoolep) [inline scripts/post-install.sh]: Can I get some detail on this permission issue?

Comment (nathanielc) [inline scripts/post-install.sh]: Binaries under /usr/bin should be owned by root not an unprivileged user,
that way only root can modify/remove then

Comment (otoolep) [inline scripts/post-install.sh]: Ah, great catch.

**Pull request merged**